admin/src/shared/CustomManyCount.tsx

```diff
@@ -0,0 +1,100 @@
+import {
+  Link,
+  PublicFieldProps,
+  InjectedFieldProps,
+  sanitizeFieldRestProps,
+} from "react-admin";
+
+import {
+  useReferenceManyFieldController,
+  useRecordContext,
+  useTimeout,
+  useCreatePath,
+} from "ra-core";
+
+import { Typography, TypographyProps, CircularProgress } from "@mui/material";
+import ErrorIcon from "@mui/icons-material/Error";
+
+export const CustomManyCount = (props: CustomManyCountProps) => {
+  const {
+    reference,
+    target,
+    filter,
+    link,
+    resource,
+    sortBy = "id",
+    sortByOrder = "ASC",
+    source = "id",
+    timeout = 1000,
+    ...rest
+  } = props;
+
+  const record = useRecordContext(props);
+  const oneSecondHasPassed = useTimeout(timeout);
+  const createPath = useCreatePath();
+
+  const { isLoading, error, total } = useReferenceManyFieldController({
+    filter,
+    page: 1,
+    perPage: 1,
+    record,
+    reference,
+    resource,
+    source,
+    target,
+    sort: { field: sortBy, order: sortByOrder },
+  });
+
+  const body = isLoading ? (
```

Comment:
not fan of the embedded ternary. can you flatten this the readibility ?
also maybe inside a <Body> component using only if for each cases, or anything better you can imagine